src/lib.rs

```diff
@@ -84,6 +84,33 @@ async fn adjust_proxied_headers(req: &mut Request<Body>) -> Result<(), Error> {
     Ok(())
 }
 
+fn host_header_match_proxy_address(host_header: String, proxy_listener: &Listen) -> bool {
+    // This function checks whether the host header matches the proxy_listen address
+    // Used for checking whether the host header was not set for HTTP1 requests
+    // as it defaults to the request destination address
+
+    // Build a listener address string, e.g. 127.0.0.1:4000
+    let listen_ip = proxy_listener.address.to_ascii_lowercase();
+    let listen_port = proxy_listener.port.to_string();
+    let mut listen_ip_joined: String = listen_ip.iter().map(|&ip| ip.to_string() + ".").collect();
+    listen_ip_joined.pop(); // remove trailing .
```

Comment:
listen_ip.join(".")

PS Majority of this code is not needed after switing to SocketAddr